tests/functional/testBenchmark.js

```diff
@@ -225,4 +225,225 @@ describe('Benchmark', function() {
+
+      // Refresh the test
+      await vpn.waitForQueryAndClick(
+        queries.screenHome.CONNECTION_INFO_RESTART.visible());
+
+      const refreshEventsList = await vpn.gleanTestGetValue("interaction", "speedTestRefresh", "main")
+      assert.strictEqual(refreshEventsList.length, 1);
+      const refreshEventExtra = refreshEventsList[0].extra;
+      assert.strictEqual("speed_test_result", refreshEventExtra.screen);
+      assert.strictEqual("select", refreshEventExtra.action);
+      assert.strictEqual("refresh", refreshEventExtra.element_id);
+
+      // Connection benchmark results loaded succesfully again
+      await vpn.waitForCondition(async () => {
+        completedEventsList = await vpn.gleanTestGetValue("impression", "speedTestResultCompleted", "main")
+        // Now the list has two events, the first one and the new one
+        return completedEventsList.length == 2;
+      });
+
+      completedEventExtra = completedEventsList[1].extra;
+      assert.strictEqual("speed_test_result", completedEventExtra.screen);
+      assert.strictEqual("impression", completedEventExtra.action);
+
+      // Close the connection benchmark
+      await vpn.waitForQueryAndClick(
+        queries.screenHome.CONNECTION_INFO_TOGGLE.visible());
+
+      const closedEventsList = await vpn.gleanTestGetValue("interaction", "speedTestClosed", "main")
+      assert.strictEqual(closedEventsList.length, 1);
+      const closedEventExtra = closedEventsList[0].extra;
+      assert.strictEqual("speed_test_result", closedEventExtra.screen);
+      assert.strictEqual("select", closedEventExtra.action);
+      assert.strictEqual("close", closedEventExtra.element_id);
+    });
+
+    it("records events while loading speed test", async () => {
+      let resolveSpeedTestResults;
+      this.ctx.downloadUrlCallback = async () => {
+        // Hold the speed test in the loading state until we are done testing the loading state.
+        await new Promise(resolve => resolveSpeedTestResults = resolve);
+      };
+
+      // Activate the VPN, otherwise the speed test feature is disabled
+      await vpn.waitForQuery(queries.screenHome.CONTROLLER_TITLE.visible());
+      await vpn.activate(true);
+
+      // Start the connection benchmark.
+      await vpn.waitForQueryAndClick(
+          queries.screenHome.CONNECTION_INFO_TOGGLE.visible());
+
+      // Connection benchmark results are loading
+      let loadingEventsList;
+      await vpn.waitForCondition(async () => {
+        loadingEventsList = await vpn.gleanTestGetValue("impression", "speedTestResultsLoading", "main")
+        return loadingEventsList.length == 1;
+      });
+
+      let loadingEventExtra = loadingEventsList[0].extra;
+      assert.strictEqual("speed_test_loading", loadingEventExtra.screen);
+      assert.strictEqual("impression", loadingEventExtra.action);
+
+      // TODO (VPN-2859): Enable this part of the test.
```

Comment:
The bug described in VPN-2859 blocks this part of the test - is that what is happening?

Reply:
That bug makes it so that the following assertions do not pass. 

So, here we are checking if the close event has the expected `screen` extra key. But that bug flashes the results screen when closed, so the `screen` extra key on the close event always has the value `speed_test_result` regardless which screen it was clicked from.

Reply:
Can you include your last couple sentences in the comment, so it's super-clear what it's waiting on?